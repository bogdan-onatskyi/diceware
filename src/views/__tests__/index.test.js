import React from 'react';

import Index from '../index';
import passwordObject from '../../models/password-object';

describe("Index component", () => {
    it("should match saved snapshot", () => {
        passwordObject.wordArray.forEach((wordObject) => {
            wordObject.index = 0;
        });
        const stores = {passwordObject};

        const wrapper = shallow(
            <Index/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should not have any props", () => {
        expect(Index.propTypes).toBe(undefined);
    });
});