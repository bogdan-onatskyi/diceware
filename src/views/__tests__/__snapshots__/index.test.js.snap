// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index component should match saved snapshot 1`] = `
<Provider
  passwordObject={
    Password {
      "_maxWords": 8,
      "_usedWords": 5,
      "caps": 63,
      "handleResetAllWords": [Function],
      "handleUsedWords": [Function],
      "init": [Function],
      "isCAPS": [Function],
      "isVariantsOpened": false,
      "separatedPassword": [Function],
      "separators": Array [
        "-",
        "_",
        ":",
        ".",
        "",
        " ",
      ],
      "toggleCAPS": [Function],
      "toggleVariantsOpened": [Function],
      "wordArray": Array [
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 1,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        },
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 2,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        },
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 3,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        },
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 4,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        },
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 5,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        },
      ],
    }
  }
>
  <div
    className="app"
  >
    <Header
      text="Генератор diceware паролей"
    />
    <Menu />
    <_class
      passwordObject={
        Password {
          "_maxWords": 8,
          "_usedWords": 5,
          "caps": 63,
          "handleResetAllWords": [Function],
          "handleUsedWords": [Function],
          "init": [Function],
          "isCAPS": [Function],
          "isVariantsOpened": false,
          "separatedPassword": [Function],
          "separators": Array [
            "-",
            "_",
            ":",
            ".",
            "",
            " ",
          ],
          "toggleCAPS": [Function],
          "toggleVariantsOpened": [Function],
          "wordArray": Array [
            Word {
              "_filter": "",
              "_index": 0,
              "_indexFiltered": Array [],
              "_wordViewId": 1,
              "editorOpened": false,
              "handleBackspace": [Function],
              "handleClick": [Function],
              "handleCountWords": [Function],
              "handleFilter": [Function],
              "handleWheel": [Function],
              "toggleEditor": [Function],
            },
            Word {
              "_filter": "",
              "_index": 0,
              "_indexFiltered": Array [],
              "_wordViewId": 2,
              "editorOpened": false,
              "handleBackspace": [Function],
              "handleClick": [Function],
              "handleCountWords": [Function],
              "handleFilter": [Function],
              "handleWheel": [Function],
              "toggleEditor": [Function],
            },
            Word {
              "_filter": "",
              "_index": 0,
              "_indexFiltered": Array [],
              "_wordViewId": 3,
              "editorOpened": false,
              "handleBackspace": [Function],
              "handleClick": [Function],
              "handleCountWords": [Function],
              "handleFilter": [Function],
              "handleWheel": [Function],
              "toggleEditor": [Function],
            },
            Word {
              "_filter": "",
              "_index": 0,
              "_indexFiltered": Array [],
              "_wordViewId": 4,
              "editorOpened": false,
              "handleBackspace": [Function],
              "handleClick": [Function],
              "handleCountWords": [Function],
              "handleFilter": [Function],
              "handleWheel": [Function],
              "toggleEditor": [Function],
            },
            Word {
              "_filter": "",
              "_index": 0,
              "_indexFiltered": Array [],
              "_wordViewId": 5,
              "editorOpened": false,
              "handleBackspace": [Function],
              "handleClick": [Function],
              "handleCountWords": [Function],
              "handleFilter": [Function],
              "handleWheel": [Function],
              "toggleEditor": [Function],
            },
          ],
        }
      }
    />
    <Footer />
  </div>
</Provider>
`;
